tinygrad/runtime/ops_gpu.py

```diff
@@ -9,26 +9,24 @@
 from tinygrad.codegen.cstyle import CStyleCodegen, CStyleLanguage
 
 OSX_TIMING_RATIO = (125/3) if OSX else 1.0   # see test/external_osx_profiling.py to determine this ratio. it's in like GPU clocks or something
-FLOAT16 = getenv("FLOAT16", 0)
 
 # TODO: if you fork and exit the child process after creating anything with cl on AMD, it hangs on e.wait()
 ROCM_LLVM_PATH = pathlib.Path("/opt/rocm/llvm/bin")
 #ROCM_LLVM_PATH = pathlib.Path(__file__).parent.parent.parent.parent / "extra/rocm/build/llvm-project/bin"
 if DEBUG >= 5:
-  from extra.helpers import enable_early_exec
-  early_exec = enable_early_exec()
+  early_exec = fromimport("extra.helpers", "enable_early_exec")()
 
 class _CL:
   def __init__(self):
     platforms: List[List[cl.Device]] = [y for y in ([x.get_devices(device_type=cl.device_type.GPU) for x in cl.get_platforms()] + [x.get_devices(device_type=cl.device_type.CPU) for x in cl.get_platforms()]) if len(y)]
-    if DEBUG >= 1: print(f"using {platforms[getenv('CL_PLATFORM', 0)]}")
-    self.cl_ctx: cl.Context = cl.Context(devices=[x for x in platforms[getenv('CL_PLATFORM', 0)] if x.name not in getenv('CL_EXCLUDE', "").split(",")])
+    if DEBUG >= 1: print(f"using devices: {[d.hashable_model_and_version_identifier for d in platforms[getenv('CL_PLATFORM', 0)] if d.name not in getenv('CL_EXCLUDE', '').split(',')]}")
```

Comment:
This line is duplicated logic from 23

Reply:
I could pull out the filtered list of devices for the current platform to a variable, but would require adding another LOC.

I could also move the log to after the context creation, but that would mean it wouldn't log the devices used to create it in the case that context creation raises an exception.

Sorry for all the hassle on this tiny PR; I'm quite new to this project and I'm not sure how far the low-LOC preference goes.

Reply:
LOC doesn't matter!!

What's the most readable?

Reply:
In that case, I'd definitely go for pulling it out to a variable:

```py
    platforms: List[List[cl.Device]] = [y for y in ([x.get_devices(device_type=cl.device_type.GPU) for x in cl.get_platforms()] + [x.get_devices(device_type=cl.device_type.CPU) for x in cl.get_platforms()]) if len(y)]
    devices: List[cl.Device] = [x for x in platforms[getenv('CL_PLATFORM', 0)] if x.name not in getenv('CL_EXCLUDE', '').split(',')]
    if DEBUG >= 1: print(f"using devices: {[d.hashable_model_and_version_identifier for d in devices]}")
    self.cl_ctx: cl.Context = cl.Context(devices=devices)
```

I'll push that up.